src/scripts/message/MessageProvider.js

```diff
@@ -0,0 +1,57 @@
+//Anna Moore
+//the goal of this module is interact with the json database 
+//provides the necessary messages array with useMessages(), getMessages()
+//provides function to alter the database saveMessage(), and deleteMessage() 
+//contains event listeners to ensure this happens at the correct time
+
+
+const eventHub = document.querySelector("#container")
+
+//intialize empthy array
+let messages = []
+
+//gather the message
+export const GetMessages = () => {
+    return fetch ('http://localhost:8088/messages')
+    .then(response =>response.json())
+    .then(parsedMessages => {
+        messages = parsedMessages
+    })
+}
+
+//provides a copy of the messages array 
+export const UseMessages = () => messages.slice()
+
+//saves the message to the json database
+export const SaveMessage = message => {
+    return fetch('http://localhost:8088/messages', {
+        method: "POST",
+        headers: {
+            "content-Type": "application/json"
+        },
+        body: JSON.stringify(message)
+    })
+    .then(GetMessages)
+    .then(dispatchMessageChangeEvent)
+}
+
+export const DeleteMessage = (messageId) => {
+    return fetch(`http://localhost:8088/messages/${messageId}`, {
+        method: "DELETE",
+    })
+    .then(GetMessages)
+    .then(dispatchMessageChangeEvent)
+  }
+
+  //update 
+  eventHub.addEventListener("deleteMessage", e => {
+      deleteMessage(e.detail.chosenMessageId)
+  })
+
+  //this purpose of thus function is??
+  const dispatchMessageChangeEvent = () => {
```

Comment:
To answer the question above.

The function `dispatchMessageChangeEvent `creates a new `CustomEvent `that is dispatching the `"MessageStateChanged"`
to the event hub. You are calling this on the functions that add delete or change your messages so when a change is made the list component can `render` the new / updated data.  It looks like you have this method here but you are not listening for it anywhere. I would listen for it on message list to render your updated messages each time instead of trying to call the `MessageList` function again which is causing a doubling up of the `GetMessages` function from the JSON-server

Reply:
This is an example of code from the `EventList` module to listen for if something was changed in the API.

```
eventHub.addEventListener("eventStateChanged", (customEvent) => {
  let events = UseEvents();
  render(events);
});
```

This will give the `MessageList` function the sole purpose of getting the initial `fetch` from the API and rendering it to the DOM on page load.